Python/Product/TestAdapter.Executor/Pytest/TestDiscovererPytest.cs

```diff
@@ -25,52 +41,94 @@ internal class TestDiscovererPytest : IPythonTestDiscoverer {
 
         public void DiscoverTests(IEnumerable<string> sources, IMessageLogger logger, ITestCaseDiscoverySink discoverySink) {
             _logger = logger;
+            string json = null;
 
-            var discoveryResults = new List<PytestDiscoveryResults>();
+            var workspaceText = _settings.IsWorkspace ? Strings.WorkspaceText : Strings.ProjectText;
+            LogInfo(Strings.PythonTestDiscovererStartedMessage.FormatUI(PythonConstants.PytestText, _settings.ProjectName, workspaceText,  _settings.DiscoveryWaitTimeInSeconds));
 
             try {
                 var env = InitializeEnvironment(sources, _settings);
                 var arguments = GetArguments(sources);
-
-                using (var outputStream = new MemoryStream())
-                using (var writer = new StreamWriter(outputStream, encoding: new UTF8Encoding(true), 4096, leaveOpen: true))
-                using (var proc = ProcessOutput.Run(
+                json = ExecuteProcess(
```

Comment:
moved processOutput code to its own function for both pytest and unittest discoverers.. logging is separated out to this level though.  New timeout error message occurs on TimeoutException